test/Constraints/diagnostics.swift

```diff
@@ -444,8 +444,7 @@ let _ : Color = .rainbow  // expected-error {{member 'rainbow' is a function; di
 let _: Color = .overload(a : 1.0)  // expected-error {{cannot convert value of type 'Double' to expected argument type 'Int'}}
 let _: Color = .overload(1.0)  // expected-error {{ambiguous reference to member 'overload'}}
 // expected-note @-1 {{overloads for 'overload' exist with these partially matching parameter lists: (a: Int), (b: Int)}}
-let _: Color = .overload(1)  // expected-error {{ambiguous reference to member 'overload'}}
-// expected-note @-1 {{overloads for 'overload' exist with these partially matching parameter lists: (a: Int), (b: Int)}}
```

Comment:
So, this bit could actually be a regression in practice, because notes in the same file are shown more nicely in Xcode than notes in other files, and especially if the candidates are imported and thus don't have *any* file. I know that's a lousy thing to make a consideration, but it may still be interesting.

Reply:
Should the notes be attached to the source location of the call then?

Reply:
Similar thing happens with requirement diagnostics at the moment - they'd get attach to the declaration of the requirement which as you mention might not be as good...

Reply:
Well, I wouldn't want to issue separate notes at the call site either. If we can't combine them into one message like before, I guess we should just take the hit. If we *can,* though, we could conditionalize on "are any of the overloads in other modules" or "do the diagnostic options have 'editor mode' on" or both.

Reply:
I don't think we can because that could be all sorts of different problems e.g. requirement failure, availability, labeling, type mismatch etc. on different overloads...